analyzer/src/main/resources/scripts/init.gradle

```diff
@@ -353,6 +351,22 @@ class AbstractDependencyTreePlugin<T> implements Plugin<T> {
             }
         }
 
+        private static boolean canBeResolved(org.gradle.api.artifacts.Configuration configuration) {
+            // The DeprecatableConfiguration class is available since Gradle 6.0.
+            boolean isDeprecatedForResolving = false
+            if (configuration.metaClass.respondsTo(configuration, "getResolutionAlternatives")) {
+                isDeprecatedForResolving = configuration."getResolutionAlternatives"() != null
+            }
+
+            // The canBeResolved property is available since Gradle 3.3.
+            boolean canBeResolved = true
```

Comment:
Couldn't you stick to the original one-liner?

    boolean canBeResolved = !configuration.hasProperty('canBeResolved') || configuration.canBeResolved